MovingBlocks/Terasology
# fix: Reloading .ui file for screens during game

### Contains

Fixes #3816 which caused UI screens to not close with ESC key after changing their .ui file mid game. Instead of closing, a null pointer exception would occur because the new screen class which was created while reloading was not associated with the NUIManager object. This was due to the fact that the class was instantiated outside of the manager, instead of being handled by it. 

### How to test

Use a module like TutorialNUI. Create a screen, and then load it using showScreen 'nameOfScreen'. Hit ESC, and the screen will close normally. Now, go to nameOfScreen.ui, and change a property of the screen and save it. Loading the screen works and you will see your changes. Finally, if you hit ESC, the screen should close as expected. 

### Outstanding before merging

This change will work if the screen is in some way a subclass of CoreScreenLayer. The class needs to have the getManager and setManager methods. If their are other UI classes which are independent of that class, they will need to be handled as well.
